src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1769,13 +1782,13 @@ export default class AircraftModel {
     _calculateTargetedAltitudeVnav() {
         const hardRestrictedWaypointModel = this.fms.nextHardAltitudeRestrictedWaypoint;
 
-        if (_isNil(hardRestrictedWaypointModel)) {
-            return;
-        }
-
         switch (this.flightPhase) {
             case FLIGHT_PHASE.TAKEOFF:
             case FLIGHT_PHASE.CLIMB:
+                if (_isNil(hardRestrictedWaypointModel)) {
+                    return;
+                }
```

Comment:
The whole thing needs refactoring, but there isn't any bugs open against the climb behaviour as the logic on line 1187 causes it to climb to the intended ceiling anyway I believe.

Reply:
I'm away from a usable comptuer at the moment... I guess my concern is what would happen if we had a SID that contained only soft restrictions? Would it comply with those restrictions as it climbed up to the `initial_altitude`? Or would it not recognize there being any restrictions since none of them are _hard restrictions_, and just blast straight up for the initial_alt?

Setting one SID to having various nonsensical altitude restrictions and the aircraft to having a climb rate of like 50,000fpm should get us the answer. :wink:

Reply:
I suspect it doesn't behave correctly, but it does probably behave playably at the moment ;)  I'll see about writing a couple of tests for the refactor.